includes/User.php

```diff
@@ -4851,11 +4851,15 @@ protected function saveOptions() {
 		if ($wgPreferencesUseService) {
 			$preferenceNames = array_keys($this->userPreferences()->getPreferences($this->getId()));
 
-			(new WikiaSQL())
+			$sql = (new WikiaSQL())
 				->DELETE('user_properties')
-				->WHERE('up_user')->EQUAL_TO($this->getId())
-					->AND_('up_property')->NOT_IN($preferenceNames)
-				->run($dbw);
+				->WHERE('up_user')->EQUAL_TO($this->getId());
+
+			if (!empty($preferenceNames)) {
+				$sql->AND_('up_property')->NOT_IN($preferenceNames);
+			}
+
+			$sql->run($dbw);
```

Comment:
Let the queries hit the floor!
![let the queries hit the floor](http://i0.kym-cdn.com/entries/icons/original/000/017/848/Screenshot_70.png)